langroid/agent/chat_document.py

```diff
@@ -29,7 +29,12 @@ class ChatDocMetaData(DocMetaData):
     block: None | Entity = None
     sender_name: str = ""
     recipient: str = ""
-    usage: int = 0
+    usage: dict[str, float] = {
```

Comment:
This occurs in a couple of places. Pls create a Pydantic class TokenCost with all fields except `total_tokens`, and make `total_tokens` a `@property` that simply returns the sum.

Reply:
Just to make sure we are on the same page:
1. Create the class `TokenCost` under `langroid/language_models/base.py`. 
```
class TokenCost(BaseModel):
    prompt_tokens: float
    completion_tokens: float
    cost: float
``` 
2. Use this class for the field `use` under the classes `ChatDocMetaData` and `LLMResponse`.

Reply:
yes, maybe call it `LLMTokenUsage`